### ISSUE TITLE
Schedule not triggered with condition (binary sensor)

### ISSUE THREAD

**Issue by zbuh** [nielsfaber/scheduler-component#63]
![Screenshot 2020-11-26 at 09 33 39](https://user-images.githubusercontent.com/190734/100334854-3725f380-2fcc-11eb-96ca-2217f8c6c967.png)
![Screenshot 2020-11-26 at 09 33 29](https://user-images.githubusercontent.com/190734/100334858-38572080-2fcc-11eb-94b0-412a66d16a76.png)
![Screenshot 2020-11-26 at 09 32 17](https://user-images.githubusercontent.com/190734/100334859-38efb700-2fcc-11eb-87dc-83fb05529c45.png)

Error found in the log:

```
Logger: homeassistant.components.websocket_api.http.connection
Source: helpers/script.py:926 
Integration: Home Assistant WebSocket API (documentation, issues) 
First occurred: 8:19:42 AM (3 occurrences) 
Last logged: 9:35:33 AM

[139988190400368] Error handling message: Unknown error
[139988070203552] Error handling message: Unknown error
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 18, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 49, in websocket_search_related
    msg["id"], searcher.async_search(msg["item_type"], msg["item_id"])
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 89, in async_search
    getattr(self, f"_resolve_{search_type}")(search_id)
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 167, in _resolve_entity
    for entity in automation.automations_with_entity(self.hass, entity_id):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 141, in automations_with_entity
    return [
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 144, in <listcomp>
    if entity_id in automation_entity.referenced_entities
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 327, in referenced_entities
    referenced = self.action_script.referenced_entities
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 926, in referenced_entities
    referenced.add(entity_id)
TypeError: unhashable type: 'dict'
```

**Comment by nielsfaber** [comment]
The error log you provided does not seem to be related to scheduler.

**Comment by zbuh** [comment]
Let me try again

**Comment by zbuh** [comment]
It must be. Occurred again just now with a new test

```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 18, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 49, in websocket_search_related
    msg["id"], searcher.async_search(msg["item_type"], msg["item_id"])
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 89, in async_search
    getattr(self, f"_resolve_{search_type}")(search_id)
  File "/usr/src/homeassistant/homeassistant/components/search/__init__.py", line 167, in _resolve_entity
    for entity in automation.automations_with_entity(self.hass, entity_id):
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 141, in automations_with_entity
    return [
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 144, in <listcomp>
    if entity_id in automation_entity.referenced_entities
  File "/usr/src/homeassistant/homeassistant/components/automation/__init__.py", line 327, in referenced_entities
    referenced = self.action_script.referenced_entities
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 926, in referenced_entities
    referenced.add(entity_id)
TypeError: unhashable type: 'dict'
```

![Screenshot 2020-11-26 at 10 05 39](https://user-images.githubusercontent.com/190734/100337334-25921b00-2fcf-11eb-9bfe-74fb0582ac58.png)

**Comment by nielsfaber** [comment]
Ok, but I cannot really help you then.
The error log refers to errors in components `websocket_api`, `search` and `automation`.
But none of these are used by the scheduler-component.

There are no details regarding function / line number where scheduler fails.
So I don't know where to look.

**Comment by zbuh** [comment]
Tried with a different entity, the first one was from deconz. 
Now with a zwave entity, the error in the log does not show up.
The schedule triggers as before. but the action is never done (light is not turned on)

**Comment by nielsfaber** [comment]
I guess this is due to your rain sensor never having state 'open'.
You addressed this issue before.

**Comment by zbuh** [comment]
Thought it was only a UI issue, and the test was done boolean: True/False
Should I close this issue? or can it be used for that?

**Comment by nielsfaber** [comment]
As you can see in the entity attributes, it is stored as such as well.
So although it is a bug in the UI, it leads to a wrong configuration written to the component.

Let's use this issue for follow-up on this bug.

Regarding your earlier problems with the deconz device, I don't know what's wrong.
You could try to manually call the service with the same arguments.
It should have the same effect as scheduler would have.
I am planning to add a 'test action' button in the card, for quick-checking such problems.

**Comment by nielsfaber** [comment]
I improved the conditions implementation in v3.0.0.
Please consider updating, and test if this resolves your problem.

**Comment by nielsfaber** [comment]
Closing due to lack of response..
